crates/kubelet/src/node.rs

```diff
@@ -132,21 +128,19 @@ async fn update_lease(node_uid: &str, client: &APIClient) {
 ///
 /// TODO: A lot of the values here are faked, and should be replaced by real
 /// numbers post-POC.
-fn node_definition(arch: &str) -> serde_json::Value {
-    let pod_ip = "10.21.77.2";
-    let port = 3000;
+fn node_definition(config: Config) -> serde_json::Value {
```

Comment:
Could we use the typed object from the API for a `Node` here?

https://github.com/Arnavion/k8s-openapi/blob/05e60741985f5a5c3c5da255fee93aec16157ed7/src/v1_17/api/core/v1/node.rs#L3-L14

(Not blocking this PR though.)

Reply:
This personally reads a little cleaner as is instead of having to deal with converting and cloning strings

Reply:
It's fine for now, and there's no point in holding this PR.
But if we're using the typed library, we should use it consistently.